src/main/java/org/teamresistance/frc/Robot.java

```diff
@@ -34,8 +41,13 @@ public void robotInit() {
     final SwitchReactor reactor = Strongback.switchReactor();
 
     // Hold the current angle of the robot while the trigger is held
-    reactor.onTriggeredSubmit(leftJoystick.getTrigger(), () -> new HoldAngleCommand(drive, 90));
-    reactor.onUntriggeredSubmit(leftJoystick.getTrigger(), () -> Command.cancel(drive));
+   // reactor.onTriggeredSubmit(leftJoystick.getTrigger(), () -> new HoldAngleCommand(drive, 90));
+   // reactor.onUntriggeredSubmit(leftJoystick.getTrigger(), () -> Command.cancel(drive));
+    reactor.whileTriggered(rightJoystick.getTrigger(), () -> new SnorfleOutCommand( snorfSpeed,snorfleMotor,stopSnorf));
+    reactor.whileUntriggered(rightJoystick.getTrigger(), () -> Command.cancel(snorfleMotor));
```

Comment:
Un problema.

Imagine the case where you're holding the trigger to reverse, then press 6 (to spin forward), then let go of the reverse trigger. The final state of the snorfler is stopped when it should be snorfling forward still.

The unfortunate side-effect of using a purely Command-based approach is it's hard to conditionally dispatch an event without the code becoming totally unreadable (think of `Switch` composition with instance fields of boolean flags). We've been able to skirt around this problem by having the conditional `if` checks run *after* the command has been dispatched, during the *execute* method. This workaround works for the gear grabber, but not the snorfler.

With a continuously-running process such as snorfling, by the time a dispatched Command realizes "oh shit, I shouldn't have been dispatched" it's too late to go back because the newly-dispatched Command will have already canceled whatever had been running before it.

I can't think of a great solution.

**Do you have any ideas? Gonna ping @ShreyaRavi. I think our *best* solution is to apply what we've been doing with the gear grabber:**

1. Create a `Snorfler` class that holds the motor and has methods for stopping, reversing, and snorfling forward and holds an enum with Snorfler's state (STOPPED, SNORLFING, REVERSING).
2. Create a StopSnorfling command that executes `Snorfler#stop` and use that instead of `Command.cancel`.
3. Inside `Snorfler#stop`, do the `if` check to see if we should really stop the motor.